src/react-collapsed.js

```diff
@@ -86,6 +86,8 @@ export default function useCollapse(initialConfig = {}) {
       // it's safe to clear all style overrides
       if (height === styles.height) {
         setStyles({});
+      } else {
```

Comment:
Let's add a comment describing why this `else` block is needed. Something like this would be good:

```suggestion
      } else {
      // If the heights don't match, this could be due the height of the content changing mid-transition
      // If that's the case, re-trigger the animation to animate to the new height
```